### PR TITLE
insights: enqueue retention work from a separate goroutine

### PR DESCRIPTION
**Pull request by leonore**
part of #46082 and follow up on convo 

## Test plan

Change interval and watch work queued up

### CODE REVIEW

File: enterprise/internal/insights/background/retention_enqueuer.go [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+				if err != nil {
+					return errors.Wrap(err, "unable to fetch series for retention")
+				}
+				for _, series := range allSeries {
+					_, err = retention.EnqueueJob(ctx, workerBaseStore, &retention.DataRetentionJob{SeriesID: series.ID})
+					if err != nil {
+						observationCtx.Logger.Error("could not enqueue data retention job", log.Int("seriesID", series.ID), log.Error(err))
+					}
+				}
+				return nil
```
Comment (chwarwick) [inline enterprise/internal/insights/background/retention_enqueuer.go:35]: I wasn't sure here if we would want to build up a multierror for each enqueue fail and return it or just log and ignore.  I don't think there is any retry on these

**Review by chwarwick — APPROVED** [review]

File: enterprise/internal/insights/background/retention_enqueuer.go (hunk 1 above)
Comment (leonore) [reply enterprise/internal/insights/background/retention_enqueuer.go:35]: I'll follow the insight enqueuer logic and build up a multi error

File: enterprise/internal/insights/background/retention_enqueuer.go [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+					return errors.Wrap(err, "unable to fetch series for retention")
+				}
+				var multi error
+				for _, series := range allSeries {
+					_, err = retention.EnqueueJob(ctx, workerBaseStore, &retention.DataRetentionJob{SeriesID: series.ID})
+					if err != nil {
+						multi = errors.Append(multi, errors.Wrapf(err, "seriesID: %d", series.ID))
+					}
+				}
+				if multi != nil {
```
Comment (chwarwick) [inline enterprise/internal/insights/background/retention_enqueuer.go:34]: nit: isn't this equivalent to `return multi`

Comment (leonore) [reply enterprise/internal/insights/background/retention_enqueuer.go:34]: yes, I was just being verbose in go style, I'll simplify it

**Pull request merged**